x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/local/EsqlProject.java

```diff
@@ -65,8 +66,12 @@ public EsqlProject replaceChild(LogicalPlan newChild) {
     @Override
     public boolean expressionsResolved() {
         for (NamedExpression projection : projections()) {
-            // don't call dataType() - it will fail on UnresolvedAttribute
-            if (projection.resolved() == false && projection instanceof UnsupportedAttribute == false) {
+            // Pass through for unsupported types.
+            if (projection instanceof UnsupportedAttribute
+                || (projection instanceof Alias a && a.child() instanceof UnsupportedAttribute)) {
+                continue;
+            }
+            if (projection.resolved() == false) {
```

Comment:
@astefan this fixes validation for the query `FROM sample_data, sample_data_str | rename client_ip as foo | sort bla` (very good find!).

If we're happy with the current approach, I'll add a verifier test for it.